cli/flox/src/commands/install.rs

```diff
@@ -222,6 +227,25 @@ impl Install {
 
         let mut environment = concrete_environment.into_dyn_environment();
 
+        // Get a list of the packages that this environment is already overriding via composition.
+        let lockfile_path = environment.lockfile_path(&flox)?;
+        let canonicalized_lockfile_path_res = CanonicalPath::new(lockfile_path);
+        let existing_composer_package_overrides = if let Ok(path) = canonicalized_lockfile_path_res
+        {
+            let lockfile_content =
+                std::fs::read_to_string(&path).context("Failed to read lockfile")?;
+            let lockfile =
+                Lockfile::from_str(&lockfile_content).context("Failed to parse lockfile")?;
```

Comment:
suggestion nonblocking: I think we should expose `CoreEnvironment::existing_lockfile` rather than re-implementing it

Reply:
I'll do that in a followup.